### PR TITLE
[flutter_tools] fix flutter create -t skeleton

### PR DESCRIPTION
**Pull request by christopherfujino**
Fixes https://github.com/flutter/flutter/issues/139138

This had been broken since https://github.com/flutter/flutter/pull/130090 merged, however, the test happened run with flutter_tools/pubspec.yaml in the current working directory.

### CODE REVIEW

File: packages/flutter_tools/lib/src/localizations/gen_l10n.dart [hunk 1]
```diff
@@ -48,7 +48,7 @@ Future<LocalizationsGenerator> generateLocalizations({
   precacheLanguageAndRegionTags();
 
   // Use \r\n if project's pubspec file contains \r\n.
-  final bool useCRLF = fileSystem.file('pubspec.yaml').readAsStringSync().contains('\r\n');
+  final bool useCRLF = projectDir.childFile('pubspec.yaml').readAsStringSync().contains('\r\n');
```
Comment (eliasyishak) [inline packages/flutter_tools/lib/src/localizations/gen_l10n.dart:51]: I'm curious, if we are changing where we are reading the pubspec from, do we need to find where we are writing this pubspec as well?

Comment (eliasyishak) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:51]: Yeah i just tested the PR with the command `flutter create --template=skeleton testing_pr` on my desktop and I ended up with two directories created.
- `testing_pr`: as expected
- `build`: contains only one file called `.last_build_id`

This isn't what we're expecting right?

Comment (eliasyishak) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:51]: I'm not entirely sure if we would need to change the build system to for the create skeleton use case but perhaps we can add a clean up method in the case statement for the skeleton workflow

https://github.com/christopherfujino/flutter/blob/fix-flutter-create-with-skeleton/packages/flutter_tools/lib/src/commands/create.dart#L370-L379

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:51]: i think that is a separate issue unrelated to this regression or fix. I will file another issue.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/localizations/gen_l10n.dart:51]: Done https://github.com/flutter/flutter/issues/141287

**Review by eliasyishak — APPROVED** [review]
LGTM

**Pull request merged**